File: ServerSpace/src/main/java/DataLayer/EntListaDB.java
package DataLayer;

import EntityLayer.EntListaEntity;
import Enumerados.ProcessActionEnum;
import Framework.BaseDB;
import Framework.Inj;
import Framework.Utilidades;
import Framework.injector;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class EntListaDB extends BaseDB {

    public ArrayList<EntListaEntity> ObtenerCodigo(String codigo) {

        injector InjDB = new injector();
        ArrayList<EntListaEntity> DatoMemoria = new ArrayList<>();
        EntListaEntity en;
        try {
            InjDB.Sp("sp_EntListaObtenerCodigo");
            InjDB.Pmt_String("@Codigo", codigo, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new EntListaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public ArrayList<EntListaEntity> GetListaRelacion_Main(String codigo) {
        injector InjDB = new injector();
        ArrayList<EntListaEntity> DatoMemoria = new ArrayList<>();
        EntListaEntity en;
        try {
            InjDB.Sp("sp_ListaRelacion_Main");
            InjDB.Pmt_String("@Codigo", codigo, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new EntListaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public EntListaEntity Registrar(EntListaEntity entity) {
        Boolean State = null;
        try {
            Inj.IniciarTranssaccion(false);
            String Store = "sp_PersonaNatural_Save";
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Store = "sp_PersonaNatural_Update";
            }
            Inj.Sp(Store);
            Inj.Pmt_Integer("@ListaId", entity.getListaId(), true);
            Inj.Pmt_Integer("@CampoId", entity.getCampoId(), false);
            Inj.Pmt_String("@Codigo", entity.getCodigo(), false);
            Inj.Pmt_String("@Nombre", entity.getNombre(), false);
            Inj.Pmt_String("@Descripcion", entity.getDescripcion(), false);
            Inj.Pmt_String("@CodUsuario", entity.getCodUsuario(), false);
            Inj.Pmt_String("@FechaRegistro", Utilidades.getFechaRegistro(), false);
            Inj.Pmt_Boolean("@EstadoRegistro", entity.getEstadoRegistro(), false);
            if (entity.getAction() == ProcessActionEnum.Add.getValor()) {
                int Id = Inj.RunInsert();
                State = Id > 0;
                if (State) {
                    entity.setListaId(Id);
                }
            }
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Inj.RunUpdate();
            }

        } catch (Exception ex) {
            throw new UnsupportedOperationException("Datalater : " + ex);
        }
        return entity;
    }

}

File: Space/Space.DataLayer/EntListaDB.cs
using Framework;
using Space.EntityLayer;
using System.Data;
using System.Data.SqlClient;

namespace Space.DataLayer
{
    public class EntListaDB : BaseDB
    {

        public List<EntListaEntity> ObtenerCodigo(String codigo)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_EntListaObtenerCodigo");
            Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, codigo);

            SqlDataReader reader = Conexion.ReaderData();
            List<EntListaEntity> ListaData = new List<EntListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                EntListaEntity entity = new EntListaEntity();
                if (FillFrom<EntListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            Conexion.CloseConnection();
            return ListaData;
        }

        public List<EntListaEntity> GetListaRelacion_Main(String codigo)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_ListaRelacion_Main");
            Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, codigo);

            SqlDataReader reader = Conexion.ReaderData();
            List<EntListaEntity> ListaData = new List<EntListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                EntListaEntity entity = new EntListaEntity();
                if (FillFrom<EntListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            return ListaData;
        }

        public Int32 Save(EntListaEntity Ent)
        {
            try
            {

                String NameStore = "sp_ListaRelacion_Save";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("ListaRelacionId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.ListaId);
                Conexion.AddParemeter("EntCampoId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.EntCampoId);
                Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Codigo);
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Nombre);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);
                Ent.ListaId = Conexion.ExecuteQueryItem("ListaRelacionId");

                return Ent.ListaId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }

        public Int32 Update(EntListaEntity Ent)
        {
            try
            {

                String NameStore = "sp_ListaRelacion_Update";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("ListaRelacionId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.ListaId);
                Conexion.AddParemeter("EntCampoId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.EntCampoId);
                Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Codigo);
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Nombre);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);
                Ent.ListaId = Conexion.ExecuteQueryItem("ListaId");

                return Ent.ListaId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }

        public List<EntListaEntity> GetListaRelacionLike(String codigo, String Nombre)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_ListaGenericaSelectCodigoLike");
            Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, codigo);
            Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 100, ParameterDirection.Input, Nombre);
            SqlDataReader reader = Conexion.ReaderData();
            List<EntListaEntity> ListaData = new List<EntListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                EntListaEntity entity = new EntListaEntity();
                if (FillFrom<EntListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            Conexion.CloseConnection();
            return ListaData;
        }

    }
}
